[3.x] Add CodeQL3000 tasks

dougbu opened the pull request:
- enable CodeQL in this branch
  - use explicit tasks because this is not the repo's default branch
- add a new top-level parameter to enable CodeQL3000 in manual builds
- tag CodeQL3000 builds that do actual CodeQL3000 work
- add a tsaoptions.json file
  - initial area path value is certainly wrong = other values may be incorrect too

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/8159)

dougbu commented on .azure/pipelines/build.yaml:
```diff
@@ -210,7 +237,7 @@ jobs:
 
 # Tests
-- ${{ if eq(parameters.skip_test, false) }}:
+- ${{ if and(eq(parameters.skip_test, false), ne(variables.runCodeQL3000, 'true')) }}:
```
As-is, tests are disabled in scheduled runs even if the CodeQL3000 tasks do nothing (but are enabled). It may be possible to change this condition so that tests run when the CodeQL tasks no-op. Is that necessary❔

ReubenBond replied:
I am not quite understanding the purpose of this, is it a security measure?
Under what circumstances does CodeQL3000 do nothing (runCodeQL3000 != true)?

dougbu replied:
This is to ensure that builds running the CodeQL3000 tasks aren't shown as failures due to unrelated test issues i.e. to make it easy to see whether the CodeQL3000 reporting is working. I asked the question above because Orleans is unusual in having its regular pipeline already do scheduled builds, elsewhere I _added_ a schedule just for CodeQL3000.

The CodeQL3000 tasks are included in a build only when `runCodeQL3000` is `true`. But, as [commented on above](https://github.com/dotnet/orleans/pull/8159/files#diff-66fc07ffa2464dbb58010aa7c74405252e1802e0951c8f0240698804ff0af4b3R101), the tasks do nothing of interest when the analysis is considered up to date. In that case, the pipeline will do a regular build but not test it. I'm not positive I can wire this up differently and wanted to know how important testing is considered in official builds (considering the public scheduled builds run the same tests) before trying to do something different here.

dougbu replied:
Should mention "tests are disabled in scheduled runs" applies only to _internal_ scheduled runs. Public scheduled runs will still execute tests unconditionally.

dougbu commented on .config/tsaoptions.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "areaPath": "DevDiv\\ASP.NET Core",
```
Please let me know what area path is correct. @rafikiassumani-msft or @danmoseley may know if @benjaminpetit and @ReubenBond don't.

ReubenBond replied:
How do we find area paths? It is probably correct

dougbu replied:
This is probably wrong. Should align w/ wherever you get Policheck or CredScan bugs reported. If those aren't turned on, there's a problem in any case. I'm hoping @rafikiassumani-msft and @danmoseley chime in here.

dougbu commented:
See also [internal build #20221116.3](https://orleans.visualstudio.com/Orleans/_build/results?buildId=57486&view=results) demonstrating what happens when CodeQL is enabled in a manual build. That ran in the Orleans pipeline because the Orleans-3.x pipeline doesn't use the YAML; that's UI-based.

dougbu commented:
Rebased on 3.x and switched to the right area path. This PR should be ready to go unless your https://github.com/dotnet/orleans/pull/8159#discussion_r1025463395 comment above needs additional changes.
